src/CashuWallet.ts

```diff
@@ -327,15 +347,30 @@ class CashuWallet {
 			);
 			options?.proofsWeHave?.push(...keepProofsSelect);
 
-			const { keep, send } = await this.swap(amount, sendProofs, options);
-			const keepProofs = keepProofsSelect.concat(keep);
-			return { keep: keepProofs, send };
+			let { keep, send } = await this.swap(amount, sendProofs, options);
+			keep = keepProofsSelect.concat(keep);
+
+			// strip dleq if explicitly told so
+			if (!options?.includeDleq) {
```

Comment:
`!options?.includeDleq` also evaluates to `true`i when it is undefined. So this is not explicit `false`